Passing extra parameters in `ActiveStorage::Blob#url` to S3 Client

josegomezr opened the pull request:
### Summary


This allows calls of `ActiveStorage::Blob#url` to have more interaction with the S3 Presigner, enabling, amongst other options,
custom S3 domain URL Generation.

Closes #42488

ghiculescu commented:
Nice one, thanks for the PR. Can you please add a [changelog](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#updating-the-changelog) entry?

josegomezr commented:
My pleasure, entry added & resolved conflicts with main.

zzak commented:
@josegomezr Could you squash your commits too please? :bow:

josegomezr committed:
Passing extra parameters in `ActiveStorage::Blob#url` to S3 Client

This allows calls of `ActiveStorage::Blob#url` to have more
interaction with the S3 Presigner, enabling, amongst other options,
custom S3 domain URL Generation.

Closes #42488

josegomezr commented:
@zzak Squashed :mechanical_arm: !

josegomezr commented:
If I understood correctly the contribution guide: after the contribution is merged to main, in order to make it into the next release cycle I'll need to patch the `6-1-stable` with this changes and create a new PR?

rafaelfranca commented:
No new features are added to stable branches. Those changes will only be released in Rails 7.
